portfolio/src/main/webapp/js/background.js

```diff
@@ -0,0 +1,375 @@
+const HEIGHT_TO_WIDTH = 0.3;
+/**
+ * Runs initial set up of the background and background canvas.
+ */
+function setup() {
+  const FRAMERATE = 60;
+  const backgroundContainer = document.getElementById("background-container");
+
+  const backgroundCanvas =
+    createCanvas(backgroundContainer.clientWidth,
+      backgroundContainer.clientWidth * HEIGHT_TO_WIDTH);
+  backgroundCanvas.parent("background-container");
+
+  frameRate(FRAMERATE);
+
+  initTank();
+}
+
+let ballTank;
+/**
+ * Initializes a tank for the background.
+ */
+function initTank() {
+  const EFFECT_RAD_TO_WIDTH = 0.1;
+  const BALL_SIZE_TO_WIDTH = 0.01;
+  const NUM_BALLS = 25;
+  const INIT_BALL_VEL = 1;
+
+  const ballSize = BALL_SIZE_TO_WIDTH * width;
+  ballTank = new Tank(width, height, EFFECT_RAD_TO_WIDTH * width, ballSize);
+
+  for (let i = 0; i < NUM_BALLS; i++) {
+    const randomPos = createVector(Math.random() * ballTank.width,
+      Math.random() * ballTank.height);
+
+    const randomVel = p5.Vector.random2D();
+    randomVel.mult(INIT_BALL_VEL);
+
+    const newBall = new Ball(randomPos, randomVel);
+    ballTank.addBall(newBall);
+  }
+}
+
+/**
+ * Resizes the canvas when the window is resized.
+ */
+function windowResized() {
+  const canvasContainer = document.getElementById("background-container");
+  resizeCanvas(canvasContainer.clientWidth,
+    canvasContainer.clientWidth * HEIGHT_TO_WIDTH);
+}
+
+/**
+ * Called every draw loop; used for updating game logic
+ * and drawing to the screen.
+ */
+function draw() {
+  background(255);
+  ballTank.update();
+  ballTank.draw();
+}
+
+/**
+ * Tank that holds, updates, and draws statically-charged balls.
+ */
+class Tank {
+  /**
+   * @param {number} width
+   * @param {number} height
+   * @param {number} staticRadius
+   * @param {number} ballSize
+   */
+  constructor(width, height, staticRadius, ballSize) {
+    /**
+     * Height of the tank.
+     * @type {number} @const
+     */
+    this.height = height;
+
+    /**
+     * Width of the tank.
+     * @type {number} @const
+     */
+    this.width = width;
+
+    /**
+     * Effective radius of static effect.
```

Comment:
What is a static effect?

Reply:
When I google searched static effect I got results like static on a tv screen, which I assume isn't what you're going for.

Reply:
Yeah good point. I don't really know either and should have made this more clear. Will change to `repulsion effect`